File: CSHARP/KuantumKaosYoneticisi/KuantumKaosYoneticisi/KuantumCokusuException.cs
using System;

public class KuantumCokusuException : Exception
{
    public string PatlayanNesneID { get; private set; }

    // Hata fırlatıldığında patlayan nesnenin ID'sini mesajda gösterecek constructor.
    public KuantumCokusuException(string nesneId)
        : base($"Kuantum Çöküşü! Patlayan Nesne ID: {nesneId}")
    {
        PatlayanNesneID = nesneId;
    }
}

File: JAVA/KuantumKaosYoneticisiJava/src/exceptions/KuantumCokusuException.java
package exceptions;

public class KuantumCokusuException extends RuntimeException {
    // Patlayan nesnenin ID'sini tutan alan
    private String patlayanNesneID; 

    public KuantumCokusuException(String nesneId) {
        // Hata mesajını ana sınıfa ilet (Game Over mesajının bir parçası)
        super("Kuantum Çöküşü! Patlayan Nesne ID: " + nesneId);
        this.patlayanNesneID = nesneId;
    }
    
   
    public String getPatlayanNesneID() {
        return patlayanNesneID;
    }
}